### PR TITLE
Dipole-operator into if-statements

### PR DESCRIPTION
**Pull request by Anselmoo**
Putting the dipole-operator into the if-statements for XAS and RIXS

### CODE REVIEW

File: edrixs/solvers.py [hunk 1]
```diff
@@ -387,10 +387,9 @@ def xas_1v1c_py(eval_i, eval_n, trans_op, ominc, *, gamma_c=0.1, thin=1.0, phi=0
                 raise Exception("Unknown polarization type: ", pt)
             polvec = np.zeros(npol, dtype=np.complex)
             if pt.strip() == 'left' or pt.strip() == 'right' or pt.strip() == 'linear':
-                pol = dipole_polvec_xas(thin, phi, alpha, scatter_axis, pt)
                 if npol == 3:  # dipolar transition
-                    polvec[:] = pol
-                if npol == 5:  # quadrupolar transition
+                    polvec[:] = dipole_polvec_xas(thin, phi, alpha, scatter_axis, pt)
+                elif npol == 5:  # quadrupolar transition
                     polvec[:] = quadrupole_polvec(pol, kvec)
```
Comment (shenmidelin) [inline edrixs/solvers.py]: Here, `pol` is not defined after move `dipole_polvec_xas` into `if npol==3`

Comment (mrakitin) [inline edrixs/solvers.py]: As the `pol` variable definition was removed in line 390, it's not available in this `elif` block. Flake8 complains about it in https://travis-ci.org/NSLS-II/edrixs/builds/559976493#L1685-L1687:
```
$ flake8 --max-line-length=115
./edrixs/solvers.py:393:51: F821 undefined name 'pol'
./edrixs/solvers.py:1381:51: F821 undefined name 'pol'
```

The same applies to the change below.

File: edrixs/solvers.py [hunk 2]
```diff
@@ -1376,10 +1375,9 @@ def _xas_1or2_valence_1core(
                 print("edrixs >>> Loop over for polarization: ", it, pt, flush=True)
                 kvec = unit_wavevector(thin, phi, scatter_axis, 'in')
                 polvec = np.zeros(npol, dtype=np.complex)
-                pol = dipole_polvec_xas(thin, phi, alpha, scatter_axis, pt)
                 if npol == 3:  # Dipolar transition
-                    polvec[:] = pol
-                if npol == 5:  # Quadrupolar transition
+                    polvec[:] = dipole_polvec_xas(thin, phi, alpha, scatter_axis, pt)
+                elif npol == 5:  # Quadrupolar transition
                     polvec[:] = quadrupole_polvec(pol, kvec)
```
Comment (shenmidelin) [inline edrixs/solvers.py]: Same problem, `pol` is not defined.

File: edrixs/solvers.py (hunk 1 above)
Comment (shenmidelin) [reply edrixs/solvers.py]: Thanks! I also noted this problem a few minutes ago.

Comment (mrakitin) [reply edrixs/solvers.py]: Good catch! We need to have a better test coverage in general. But even without package-related tests, having a CI integration is pretty useful already!

Comment (shenmidelin) [reply edrixs/solvers.py]: Yes, I completely agree with that. CI is very useful. However, I will work on the test parts later after I finish several papers.

**Comment by shenmidelin** [comment]
Since this pull request has been solved, let's close #36

**Pull request closed without merging**